src/app/components/todo-list/todo-list.component.ts

```diff
@@ -1,120 +0,0 @@
-import { Component, OnInit } from '@angular/core';
-import { Todo } from '../../interfaces/todo';
-import {animate, style, transition, trigger} from '@angular/animations';
-
-
-@Component({
-  selector: 'todo-list', //how i do import in into another app html <todo-list></todo-list>
-  templateUrl: './todo-list.component.html', //template
-  styleUrls: ['./todo-list.component.scss'], //styles
-  animations: [
-    trigger('fade', [
-      transition(':enter', [
-        style({ opacity: 0, transform: 'translateY(30px)'}),
-        animate(200, style({opacity: 1, transform: 'translateY(0px)'}))
-      ]),
-      transition(':leave', [
-        animate(200, style({opacity: 1, transform: 'translateY(0px)'}))
-      ])
-    ])
-  ]
-})
-export class TodoListComponent implements OnInit {
-  todos: Todo[];
-  todoTitle: string;
-  id: number;
-  beforeEditCache: string;
-  filter: string;
-
-  constructor() { }
-
-  ngOnInit() {
-    this.todoTitle = '';
-    this.beforeEditCache = '';
-    this.filter = 'all';
-    this.todos = [
-      {
-        id: 1,
-        title: 'Finish Angular course',
-        completed: false,
-        editing: false
-      },
-      {
-        id: 2,
-        title: 'Take over the world',
-        completed: false,
-        editing: false
-      },
-      {
-        id: 3,
-        title: 'Fix js fundamentals',
-        completed: false,
-        editing: false
-      }
-    ];
-    this.id = this.todos.length - 1;
-  }
-
-  addToDo(): void {
-    if (!this.todoTitle.trim()) {
```

Comment:
it's strange that you're adding todo this way. For me it should be in a way where you pass todo elem as a parameter, then check if everything is OK you add to your storage\obj\whatever. Also, this fn shouldn't be responsible for the next: 
this.todoTitle = '';
    this.id++;

Reply:
plus, almost every fn is not pure. I don't like non-pure approach.